生成されるファイルのフォーマットを最新のコーディング規約に合わせて修正する

hassy-haruto opened the pull request:
## 実装の目的/背景
#6 のissueに基づき、コーディングチェックに引っ掛かるようになった箇所を修正しました

## テスト証跡

lancers_admin内でブランチを切り
```
bin/cake bake extended_model payments
```
コマンドを実行し、今回の修正で引っ掛からなくなることを確認

※TestCaseの自動生成ファイルはあらかじめ削除しました

- 修正前
<img src="https://user-images.githubusercontent.com/62461866/144418406-d30ba265-2682-473f-8d84-4e753f212d65.png">

- 修正後　（コマンド整形のslack通知が来ないことを確認）
<img src="https://user-images.githubusercontent.com/62461866/144418433-daabcf7c-60f7-445e-9fbe-3f5ff6f8b55c.png">

hassy-haruto committed:
バージョンアップによる整形コマンドのズレに対応

hassy-haruto commented:
## 企画部門

- 開発部

## 利用部門

- 開発部

## 実装の目的/背景
#6 のissueに基づき、コーディングチェックに引っ掛かるようになった箇所を修正しました

## 非機能要件（or 不安なこと）

## テスト証跡

lancers_admin内でブランチを切り
```
bin/cake bake extended_model payments
```
コマンドを実行し、今回の修正で引っ掛からなくなることを確認

※TestCaseの自動生成ファイルはあらかじめ削除しました

- 修正前
<img src="https://user-images.githubusercontent.com/62461866/144418406-d30ba265-2682-473f-8d84-4e753f212d65.png">

- 修正後　（コマンド整形のslack通知が来ないことを確認）
<img src="https://user-images.githubusercontent.com/62461866/144418433-daabcf7c-60f7-445e-9fbe-3f5ff6f8b55c.png">

## リリース手順

https://github.com/LancersDevTeam/lancers_admin

## リリース対象

- Lancers
  - 本番環境

## リリース前チェック項目

- [ ] [プロジェクト管理シート](http://bit.ly/2kL3ta9) の「中プロジェクト（GitHubラベル）」で管理されているLabelが付与されていること
- [ ] [リスク要因](https://goo.gl/5Bf5KP) に該当するとき、Label が付与されていること
- [ ] テスト証跡が記載されていること
- [ ] 利用部門が検証環境で動作確認していること -- 「検証環境で動作確認しました」
- [ ] Pull Request の作成者 "以外" が動作確認していること

## リリース後チェック項目

- [ ] 利用部門が本番環境でリリース確認していること --「本番環境でリリース確認しました」
- [ ] Pull Request の作成者 "以外" の利用部門がリリース確認していること

mamy1326 reviewed (approved):
おお！いいですね！非常に助かります！
対応ありがとうございますー！

コード、テストOKです。
検証環境で確認しました。

mamy1326 commented:
@hassy-haruto 

> ※TestCaseの自動生成ファイルはあらかじめ削除しました

コマンドに `--no-test` オプションをつけるとテストコード出力しなくなるので覚えておくといいかも？
```bash
bin/cake bake extended_model payments --no-test
```

実は `--help` すると答えが書いてあったりします。
`--no-test              Do not generate a test case skeleton.`

```bash
$ bin/cake bake extended_model --help
Bake table and entity classes.

Usage:
cake bake extended_model [options] [<name>]

Options:

--connection, -c       The datasource connection to get data from.
                       (default: default)
--display-field        The displayField if you would like to choose one.
--fields               A comma separated list of fields to make
                       accessible.
--force, -f            Force overwriting existing files without
                       prompting.
--help, -h             Display this help.
--hidden               A comma separated list of fields to hide.
--no-associations      Disable generating associations.
--no-entity            Disable generating an entity class.
--no-fields            Disable generating accessible fields in the
                       entity.
--no-fixture           Do not generate a test fixture skeleton.
--no-hidden            Disable generating hidden fields in the entity.
--no-rules             Disable generating a rules checker.
--no-table             Disable generating a table class.
--no-test              Do not generate a test case skeleton.
--no-validation        Disable generating validation rules.
--plugin, -p           Plugin to bake into.
--primary-key          The primary key if you would like to manually set
                       one. Can be a comma separated list if you are
                       using a composite primary key.
--quiet, -q            Enable quiet output.
--table                The table name to use if you have
                       non-conventional table names.
--theme, -t            The theme to use when baking code.
                       (default:
                       GenerationGapModelBaker)
                       (choices:
                       AdminLTE|Bake|GenerationGapModelBaker|Migrations)
--verbose, -v          Enable verbose output.

Arguments:

name  Name of the model to bake (without the Table suffix). You can use
      Plugin.name to bake plugin models. (optional)

Omitting all arguments and options will list the table names you can
generate models for.
```

hassy-haruto commented:
@mamy1326 
ありがとうございます！
久しぶりにbake生成コマンドを使ったのですっかり忘れてました。。
次回以降参考にします！

mamy1326 commented:
@hassy-haruto 
いえいえ！僕もしょっちゅう忘れるので `--help` だけ覚えてますw
PR もディスカッションだし、記録に残る形で技術の会話すると口頭で言うより自他ともに記憶に残るので書いた感じです！

hassy-haruto commented:
@mamy1326 
僕も--helpなどの詳細コマンドを確認する癖つけようと思います…笑
確かに先人方のプルリクディスカッションは参考になってました！
今回の配属先ではgithubでの会話を積極的に意識していきたいと思います！

mamy1326 commented:
@hassy-haruto 
いいですね！
細かいこと言うと、 `GitHub` などの単語の大文字小文字、単語の間にスペース入れる入れないなども気にするといいです。

なぜかというと、正式名称を正確に覚えるのは大事（自分のサービス作ったら名前とかこだわりあるはず）なんですが、こういうとこ大事にすると、プログラム書いてて変数名にもこだわるようになっていって、結果的に読みやすいコードを書けるようになるからです。

ブログ書くときのレビューポイントとしても覚えておくと自然と読みやすいものが出来上がったりするので覚えておくといいかなーと！

isanasan committed:
update: 不要なuse文を削除

下記エラーが残っていたので修正した。
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
  7 | ERROR | [x] Header blocks must not contain blank lines
  9 | ERROR | [x] Type Cake\ORM\Query is not used in this file.
  9 | ERROR | [x] Expected 0 lines between same types of use
    |       |     statement, found 1.
 11 | ERROR | [x] Type Cake\ORM\Table is not used in this file.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

isanasan commented:
# 動作確認手順
## プラグインをclone
```bash
$ cd www/lancaers_admin
$ mkdir plugins/GenerationGapModelBaker
$ git clone git@github.com:LancersDevTeam/cakephp-generation-gap-pattern-model-baker.git plugins/GenerationGapModelBaker/
$ cd plugins/GenerationGapModelBaker
```

## 本PRのブランチに切り変え
```bash
git switch 20211202_modify_plugin
```

## autoloadの設定
`composer.json`
```diff
    "autoload-dev": {
        "psr-4": {
            "App\\Test\\": "tests/",
-           "Cake\\Test\\": "vendor/cakephp/cakephp/tests/"
+           "Cake\\Test\\": "vendor/cakephp/cakephp/tests/",
+           "GenerationGapModelBaker\\": "plugins/GenerationGapModelBaker/src/"

        }
    },
```

autoloadファイルを更新
```bash
php composer.phar dumpautoload
```

## (一応)重複してるファイルを考慮して削除しておく
```bash
rm -rf src/Model/Baked/ src/Model/Entity/ src/Model/Table/
```

## コマンド実行

```bash
bash-4.2# bin/cake bake extended_model payments
One moment while associations are detected.

Baking table class for Payments...

Creating file /var/www/lancers_admin/src/Model/Baked/Table/PaymentsTable.php
Wrote `/var/www/lancers_admin/src/Model/Baked/Table/PaymentsTable.php`

Baking entity class for Payment...

Creating file /var/www/lancers_admin/src/Model/Baked/Entity/Payment.php
Wrote `/var/www/lancers_admin/src/Model/Baked/Entity/Payment.php`

Baking entended table class for Payments...

Creating file /var/www/lancers_admin/src/Model/Table/PaymentsTable.php
Wrote `/var/www/lancers_admin/src/Model/Table/PaymentsTable.php`

Baking entended entity class for Payment...

Creating file /var/www/lancers_admin/src/Model/Entity/Payment.php
Wrote `/var/www/lancers_admin/src/Model/Entity/Payment.php`

Baking test fixture for Payments...

Creating file /var/www/lancers_admin/tests/Fixture/PaymentsFixture.php
Wrote `/var/www/lancers_admin/tests/Fixture/PaymentsFixture.php`
Bake is detecting possible fixtures...

Baking test case for App\Model\Table\PaymentsTable ...

Creating file /var/www/lancers_admin/tests/TestCase/Model/Table/PaymentsTableTest.php
Wrote `/var/www/lancers_admin/tests/TestCase/Model/Table/PaymentsTableTest.php`
Done
```

## フォーマット確認
```bash
$ git add src/
$ docker run --rm -v `pwd -W`:/app isanasan/phpcs phpcs --standard=.circleci/phpcs.xml `git diff --cached --name-only` -n
```

isanasan reviewed (approved):
検証環境で動作確認しました。
コード、テスト共に良いと思います。

mamy1326 reviewed (approved):
対応ありがとうございます！みんなでやってくれて嬉しい！

コード、テストOKです。
検証環境で確認しました。

mamy1326 commented:
@isanasan 
このリポジトリは権限絞ってないのでマージしちゃいますね。

mamy1326 commented:
@hassy-haruto @isanasan 
https://packagist.org/packages/lancers/cakephp-generation-gap-pattern-model-baker
tag 1.0.2 に UPDATE されているので、各自の環境で composer update してみてくださいー

isanasan commented:
リリース確認できました
```bash
bash-4.2# php composer.phar update lancers/cakephp-generation-gap-pattern-model-baker
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 1 update, 0 removals
  - Upgrading lancers/cakephp-generation-gap-pattern-model-baker (1.0.2 => 1.0.3)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Downloading lancers/cakephp-generation-gap-pattern-model-baker (1.0.3)
  - Upgrading lancers/cakephp-generation-gap-pattern-model-baker (1.0.2 => 1.0.3): Extracting archive
Generating autoload files
73 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
phpstan/extension-installer: Extensions installed
```

isanasan commented:
close #6
